### PR TITLE
nfs: support object store and full-featured sync

### LINKED ISSUE #3702: Support NFS as object store
Similar to sftp, we can use `sync` to work with NFS directly.

### PR DESCRIPTION
**Pull request by Hexilee**
close #3702

### CODE REVIEW

File: pkg/object/file_unix.go [hunk 1]
```diff
@@ -36,6 +37,9 @@ func getOwnerGroup(info os.FileInfo) (string, string) {
 	case *sftp.FileStat:
 		owner = utils.UserName(int(st.UID))
 		group = utils.GroupName(int(st.GID))
+	case *nfs.Fattr:
```
Comment (davies) [inline pkg/object/file_unix.go:40]: the build will fail with tag !nonfs

Comment (Hexilee) [reply pkg/object/file_unix.go]: Fixed

File: pkg/sync/sync.go [hunk 2]
```diff
@@ -556,11 +555,7 @@ func worker(tasks <-chan object.Object, src, dst object.ObjectStorage, config *C
 			}
 			var err error
 			if config.Links && obj.IsSymlink() {
-				if err = copyLink(src, dst, key); err == nil {
-					copied.Increment()
-					break
-				}
-				logger.Errorf("copy link failed: %s", err)
+				err = copyLink(src, dst, key)
```
Comment (davies) [inline pkg/sync/sync.go:558]: checkSum() doe not work with symlinks, we should skip it or fix it

Comment (Hexilee) [reply pkg/sync/sync.go:559]: skipped

Comment (Hexilee) [reply pkg/sync/sync.go:559]: Is there any idea to check the copied symlink?

**Pull request merged**